doc/designs/quic-design/quic-ackm.md

```diff
@@ -0,0 +1,411 @@
+QUIC ACK Manager
+================
+
+The QUIC ACK manager is responsible for, on the TX side:
+
+  - Handling received ACK frames
+  - Generating notifications that a packet we sent was delivered successfully
+  - Generating notifications that a packet we sent was lost
+
+On the RX side, it is responsible for:
+  - Generating ACK frames for later transmission in response to packets we
+    received
+
+In order to allow it to perform these tasks, the ACK manager must:
+
+  - be notified of all transmitted packets
+  - be notified of all received datagrams
+  - be notified of all received packets
+  - be notified of all received ACK frames
+  - be notified when a packet number space is discarded
+  - be notified when its loss detection deadline arrives
+
+The ACK manager consumes:
+
+  - an arbitrary function which returns the current time;
```

Comment:
Why wouldn't the ACK manager should be raising timeout events directly?

This would make QUIC_ACKM_on_timeout and QUIC_ACKM_get_loss_detection_timeout internal only.

Reply:
It's possible, at the cost of coupling the ACK manager to the event queue. I don't mind either way.